pkg/controller/deployment/deployment_controller.go

```diff
@@ -567,9 +567,9 @@ func (dc *DeploymentController) getPodMapForDeployment(d *apps.Deployment, rsLis
 // This function is not meant to be invoked concurrently with the same key.
 func (dc *DeploymentController) syncDeployment(key string) error {
 	startTime := time.Now()
-	klog.V(4).Infof("Started syncing deployment %q (%v)", key, startTime)
+	klog.V(4).InfoS("Started syncing deployment", "deployment", key, "startTime", startTime)
```

Comment:
We should avoid passing different values into same logging key, `deployment` is reserved for using `klog.KObj` or `klog.KRef`. I would propose to either

Parse key and use KRef

```suggestion
    ns, name, _ := cache.SplitMetaNamespaceKey(key.(string))
	klog.V(4).InfoS("Started syncing deployment", "deployment", klog.KRef(ns, name), "startTime", startTime)
```

or use separate key
```suggestion
	klog.V(4).InfoS("Started syncing deployment matching cache key", "cacheKey", key, "startTime", startTime)
```

First method is more consistent with other logs, but results in more complicated code.
Second option is easier to implement, but less useful for correlating logs.

Reply:
Noticed code below, looks like key is parsed couple of lines below. We could change order of lines  too.

Reply:
changed the order of lines in favor of the first (more consistent method) and replaced key with `klog.KRef(ns, name)`.
Thanks!